### PR TITLE
Prevent writing empty cni conflist

### LINKED ISSUE #150: Kmesh CNI overwrite cni conflist with empty value
**What happened**:

After running kmesh for some time, we cannot restart kmesh daemon and also not able to start a new application pod.

By looking into kmesh logs, we do see:

```
time="2024-03-06T10:06:41Z" level=info msg="Copied /usr/bin/kmesh-cni to /opt/cni/bin." subsys="cni installer"
time="2024-03-06T10:06:41Z" level=info msg="failed to read conflist: /etc/cni/net.d/10-calico.conflist, error parsing configuration list: unexpected end of JSON input" subsys="cni installer"
time="2024-03-06T10:06:41Z" level=error msg="can not found the valid cni config!\n" subsys="cni installer"
time="2024-03-06T10:06:41Z" level=error msg="can not found the valid cni config!\n" subsys=manager
```

**What you expected to happen**:

Kmesh should not overwrite invalid conflist

**How to reproduce it (as minimally and precisely as possible)**:

not sure how to reproduce

**Anything else we need to know?**:

**Environment**:
- Kmesh version:
- Others:

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #150

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fixed kmesh cni installer write empty cni conflist when kmesh daemon is restarted.
```

### CODE REVIEW

File: pkg/cni/chained.go [hunk 1]
```diff
@@ -64,22 +64,23 @@ func getCniConfigPath() (string, error) {
...
 		var confList *libcni.NetworkConfigList = nil
 
-		for _, confFile = range files {
-			confList, err = libcni.ConfListFromFile(confFile)
+		for _, file := range files {
+			confList, err = libcni.ConfListFromFile(file)
 			if err != nil {
-				err = fmt.Errorf("failed to read conflist: %v, %v", confFile, err)
+				err = fmt.Errorf("failed to read conflist: %v, %v", file, err)
 				log.Info(err)
```
Comment (LiZhenCheng9527) [inline pkg/cni/chained.go:71]:
```suggestion
				log.Info("failed to read conflist: %v, %v", file, err)
```
Why not use one line of code to output the error log?

File: pkg/cni/chained.go [hunk 2]
```diff
@@ -112,7 +113,7 @@ func insertCNIConfig(oldconfig []byte) ([]byte, error) {
 			return nil, err
 		}
 		if plugin["type"] == kmeshCniPluginName {
-			return nil, nil
+			log.Infof("%s was installed but not cleaned up, but we would overwrite it", kmeshCniPluginName)
```
Comment (nlgwcy) [inline pkg/cni/chained.go:116]: Will append repeated？

File: pkg/cni/chained.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/cni/chained.go:71]: Tep, this is not a good way to process err.

Let's handle errors in a separate pr

File: pkg/cni/chained.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pkg/cni/chained.go:116]: You are right. THis is not intended.

**Pull request merged**